tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4310,88 +4320,88 @@ TEST_F(ValidationTest, NodeInputCompatibility) {
   RewriteAssemblyCheckMsg(
       L"..\\DXILValidation\\node_input_compatibility.hlsl", "lib_6_8",
       pArguments.data(), 2, nullptr, 0,
-      {"!13 = !{i32 1, i32 97"}, // DispatchNodeInputRecord
-      {"!13 = !{i32 1, i32 65"}, // GroupNodeInputRecords
+      {"= !{i32 1, i32 97"}, // DispatchNodeInputRecord
```

Comment:
Is there a reason this uses a different pattern to the ones above that were verifying that the thing on the left of the `=` matched?

Reply:
If by the above, you mean the pattern matching in ComputeNodeCompatibility, that one requires repeating some of the metadata references in the replacement text, so they need to be captured. In this case, we're just changing an integer on the right side of the equation. The shader is simpler and this is sufficient to match it.

Reply:
Actually, that's true of the below, but not so much the above. I suppose it could be simpler there.

Reply:
Okay, I've reminded myself that in the above case, we have several node entries that have the same metadata. To distinguish and alter them independently, we need to bring in more information including the metadata that identifies the function in question. 

Without this level of specificity, each replacement will alter the shader or launch type data for all matching disassembly. That happens to work, but it's not really what's meant to be tested.

Reply:
Did we need to copy any updates to these changes back into the mesh nodes branch?

Reply:
There are no unique updates here. These commits are all copied from those that were part of #6436. Nothing has changed. This is the copy from there to here.